src/jshint.js

```diff
@@ -2581,6 +2649,9 @@ var JSHINT = (function() {
           (isFunctor(ret) && !isEndOfExpr()) ||
           // Used as the return value of a single-statement arrow function
           (ret.id === "{" && preceeding.id === "=>") ||
+          // Used to cover a unary expression as the left-hand side of the
+          // exponentiation operator
+          (beginsUnaryExpression(ret) && state.tokens.next.id === "**") ||
```

Comment:
I'm down for that! I wrote this originally, though, so I'm at a disadvantage when it comes to identifying improvements. Maybe you can get the ball rolling in a follow-up pull request?